pkg/test/framework/components/echo/echotest/filters.go

```diff
@@ -0,0 +1,91 @@
+package echotest
+
+import "istio.io/istio/pkg/test/framework/components/echo"
+
+type (
+	sourceFilter      func(echo.Instances) echo.Instances
+	destinationFilter func(from echo.Instance, to echo.Instances) echo.Instances
+)
+
+// From applies each of the filter funcitons in order to allow removing workloads from the set of clients.
+func (t *T) From(filters ...sourceFilter) *T {
+	for _, filter := range filters {
+		t.sources = filter(t.sources)
+	}
+	return t
+}
+
+// To appends the given filters where are executed per test. Destination filters may need
+// to change behavior based on the client. For example, naked services can't be reached cross-network, so
+// the client matters.
+func (t *T) To(filters ...destinationFilter) *T {
+	t.destinationFilters = append(t.destinationFilters, filters...)
+	return t
+}
+
+// OnRegularPodSource finds the first Pod deployment that has a sidecar and doesn't use a headless service and removes all
+// other "regular" pods that aren't part of the same Service. Pods that are part of the same Service but are in a
+// different cluster or revision will still be included.
+var OnRegularPodSource sourceFilter = func(instances echo.Instances) echo.Instances {
```

Comment:
OnRegularPodSource -> OneRegularPodSource? Or the inverse below?

Also "one pod" seems a bit misleading since its many pods in one service?

Reply:
Yeah, should have said `One`. Not sure what to call this... `OneSimplePodBasedService`?